engine/source/scene_loader.cpp

```diff
@@ -12,34 +13,46 @@
 #include "resource_management/mesh_resource_manager.hpp"
 
 #include <entt/entity/entity.hpp>
+#include <glm/glm.hpp>
+#include <glm/gtx/matrix_decompose.hpp>
 #include <single_time_commands.hpp>
 
-entt::entity LoadNodeRecursive(ECS& ecs, const Hierarchy::Node& currentNode, const GPUModel& model)
+entt::entity LoadNodeRecursive(ECS& ecs, const Hierarchy::Node& currentNode, entt::entity parent, const CPUModel& cpuModel, const GPUModel& model)
 {
     const entt::entity entity = ecs.registry.create();
 
     ecs.registry.emplace<NameComponent>(entity).name = currentNode.name;
     ecs.registry.emplace<TransformComponent>(entity);
 
-    TransformHelpers::SetLocalTransform(ecs.registry, entity, currentNode.transform);
     ecs.registry.emplace<RelationshipComponent>(entity);
 
+    if (parent != entt::null)
+    {
+        RelationshipHelpers::AttachChild(ecs.registry, parent, entity);
+    }
+    TransformHelpers::SetLocalTransform(ecs.registry, entity, currentNode.transform);
+
     if (currentNode.meshIndex.has_value())
     {
         ecs.registry.emplace<StaticMeshComponent>(entity).mesh = model.meshes.at(currentNode.meshIndex.value());
+
+        TempPhysicsData tempData;
+        tempData.boundingBox = cpuModel.meshes.at(currentNode.meshIndex.value()).GetMeshBounds();
+
+        ecs.registry.emplace<TempPhysicsData>(entity, tempData);
```

Comment:
Regarding using two components for Bounding boxes and StaticMeshComponent, why do we not simply store the data along with a Rigidbody component or StaticMeshComponent

Reply:
I decided to store it in the static mesh component in the end, Rigid body component is not a good choice since we don't want to create rigid body components in the model loader.